### PR TITLE
Fix column_of behavior for missing columns

### PR DESCRIPTION
**Pull request by stackmystack**
Never throw exceptions, always return `nil` instead. This is useful for subqueries.

### CODE REVIEW

File: lib/arel_extensions/helpers.rb [hunk 1]
```diff
@@ -29,20 +29,23 @@ def self.column_of_via_arel_table(table_name, column_name)
...
+      else
+        if ActiveRecord::Base.connection.pool.respond_to?(:pool_config)
+          ActiveRecord::Base.connection.pool.pool_config.schema_cache.columns_hash(table_name)[column_name]
+        elsif ActiveRecord::Base.connection.pool.respond_to?(:schema_cache)
+          ActiveRecord::Base.connection.pool.schema_cache.columns_hash(table_name)[column_name]
+        else
+          column_of_via_arel_table(table_name, column_name)
+        end
       end
+    rescue
```
Comment (akimd) [inline lib/arel_extensions/helpers.rb:47]: You don't need the begin/end in Ruby.
```
def foo
...
rescue
...
end
```
does what you mean.

I don't think we should discard any error.  I would only catch the ones triggered by `('foo', 'bar')`.

And I'm uneasy with completely dropping errors: we never `warn` or `log` them?

Introducing `pool = ActiveRecord::Base.connection.pool` would be welcome.

File: lib/arel_extensions/math.rb [hunk 2]
```diff
@@ -40,7 +40,7 @@ def +(other)
       when Arel::Nodes::Function
         Arel.grouping(Arel::Nodes::Addition.new self, other)
       else
-        col = self.respond_to?(:relation)? ArelExtensions::column_of(self.relation.table_name, self.name.to_s) : nil
+        col = self.respond_to?(:relation)? ArelExtensions.column_of(self.relation.table_name, self.name.to_s) : nil
```
Comment (akimd) [inline lib/arel_extensions/math.rb:41]: I would personally prefer the short version, `Arel.column_of`.  But Julien might disagree with me.

Comment (akimd) [reply lib/arel_extensions/math.rb:41]: Also, that's the kind of place where, IMHO, in Ruby one would favor

```ruby
col = ArelExtensions.column_of(self.relation.table_name, self.name.to_s) if self.respond_to?(:relation)?
```

Comment (jdelporte) [reply lib/arel_extensions/math.rb]: It is defined to be used.

**Review by akimd — APPROVED** [review]
LGTM.

**Pull request merged**